code/controllers/ErrorPageControllerExtension.php

```diff
@@ -9,7 +9,8 @@
 class ErrorPageControllerExtension extends Extension {
 
 	public function onBeforeHTTPError($statusCode, $request) {
-		$response = ErrorPage::response_for($statusCode);
+		// Don't use the HTML response for media requests
+		$response = $request->isMedia() ? null : ErrorPage::response_for($statusCode);
```

Comment:
FYI: Example of missing image with full 404 response. This is using 3 branch without this PR in place. Note response time compared to found resources.

![home_ _your_site_name](https://cloud.githubusercontent.com/assets/1079425/12129575/5408a5d6-b3b9-11e5-89ea-a3bb19a606f8.png)

But even if you do a plain text response it still takes 150ms or so for me so it's not a massive saving.